### PR TITLE
Test(rpc): Test support for different api modules

### LINKED ISSUE #1224: Add more RPC module tests
### Describe the feature

With https://github.com/paradigmxyz/reth/pull/1222
we have helper functions for launching RPC server bindings via the Client traits

we need to cover every available module

## TODO

* replicate the `test_call_<module>_functions_<transport>` functions for other modules

some of them (like tracing/debug) are not implemented yet, responses should be checked for the corresponding rpc error instead.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ensi321**
Closes #1224

### CODE REVIEW

File: crates/rpc/rpc-builder/tests/it/http.rs [hunk 1]
```diff
@@ -44,43 +45,69 @@ where C: ClientT + SubscriptionClientT + Sync {
...
+    assert!(is_unimplemented(EthApiClient::hashrate(client).await.err().unwrap()));
+    assert!(is_unimplemented(EthApiClient::get_work(client).await.err().unwrap()));
+    assert!(is_unimplemented(EthApiClient::submit_hashrate(client, U256::default(), H256::default()).await.err().unwrap()));
+    assert!(is_unimplemented(EthApiClient::submit_work(client, H64::default(), H256::default(), H256::default()).await.err().unwrap()));
+    // EthApiClient::send_transaction(client).await.err().unwrap()));
+    // EthApiClient::send_raw_transaction(client).await.err().unwrap()));
+    // EthApiClient::sign(client).await.err().unwrap()));
+    // EthApiClient::sign_transaction(client).await.err().unwrap()));
+    // EthApiClient::sign_typed_data(client).await.err().unwrap()));
+    // EthApiClient::get_proof(client).await.err().unwrap()));
```
Comment (mattsse) [inline crates/rpc/rpc-builder/tests/it/http.rs]: why are they currently commented?

**Review by mattsse — CHANGES_REQUESTED** [review]
this is great! q regarding the commented calls

It is useful to have either
* test the endpoint is working
* or returns an unimplemented error so the node does not panic

File: crates/rpc/rpc-builder/tests/it/http.rs (hunk 1 above)
Comment (gakonst) [reply crates/rpc/rpc-builder/tests/it/http.rs]: @mattsse maybe you were looking at an older version of the PR?

**Review by gakonst — APPROVED** [review]
LGTM, and as we implement more calls we should turn these tests on.

@mattsse latest version of the PR seesm to have no uncommented code, so good by me

**Pull request merged**